bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java

```diff
@@ -119,4 +120,13 @@ public void setStrandLocal(String key, Object value) {
     public Object getStrandLocal(String key) {
         return strand.getProperty(key);
     }
+
+    /**
+     * Gets the property map stored in the strand.
+     *
+     * @return property map stored in the strand.
+     */
+    public Map<String, Object> getStrandLocalProperties() {
```

Comment:
FYI; When a new strand is created, there are instances where we need to propagate the property map. For instance, during async invocation(https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java#L96), we do not pass the parent strand to the scheduler. Hence there's no means of propagating the properties. Unless we expose the property map so that it can be passed during async invocation, we need to extract the properties calling getStrandLocal() passing each key.

Reply:
@pcnfernando I am afraid StrandLocal will become a kitchen sink people will through everything at. Can we please take a look the reason you needed this in the async invocation. Shall we schedule a meeting.

Reply:
Had a chat with @manuranga and decided not to expose the property map, but to create a submap with needed properties to propagate. For the moment, we'll be passing transaction context alone. @chamil321, please  note